site/src/routes/embedding-knn/AgreementClusterStaticPlot.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
+  import { browser } from "$app/environment";
+
+  export let data: any[] = [];
+  let selected: string = "normoc";
+  let api_url: string =
+    "/api/v1/data/processed/birdclef-2022/birdnet-embeddings-with-neighbors-static/v1";
+
+  $: names = data.map(async (d) => {
+    if (browser) {
+      let resp = await fetch(
+        "/api/v1/data/processed/birdclef-2022/birdnet-embeddings-with-neighbors-static/v1/" +
+          d["ego_primary_label"] +
+          "/distances.png"
+      );
+      if (resp.status != 404) {
+        return d["ego_primary_label"];
+      }
+    }
+  });
+</script>
+
+<div class="selection">
+  <b>Species:</b>
+  {#each names as name}
+    {#await name then name}
+      {#if name != undefined}
```

Comment:
IMO this is maybe too defensive, unless there was some weird issue that cropped up for you?

Reply:
With the approach I took above, what I got was an array of Promises so the #await was to unwrap the Promise. Since the arrow function was async (to accommodate for the fetch), it would return a Promise for species that didn't have images as well so I needed to filter out the species which didn't have images (which would be undefined)